src/main/java/com/almightyalpaca/discord/jdabutler/util/MiscUtils.java

```diff
@@ -69,4 +75,21 @@ public static String hastebin(final String text)
             return null;
         }
     }
+
+    public static void announce(TextChannel channel, Role role, Message message, boolean slowmode)
+    {
+        CompletionStage<?> base;
+        if (slowmode)
+            base = channel.getManager().setSlowmode(30).submit();
+        else
+            base = CompletableFuture.completedFuture(null);
+
+        CompletionStage<?> announcement = base
+                .thenRun(() -> role.getManager().setMentionable(true).submit()
+                        .thenRun(() -> channel.sendMessage(message).submit()
+                                .thenRun(() -> role.getManager().setMentionable(false).submit())));
```

Comment:
Isn't there a way of chaining those in a way where you don't have to go into multiple levels?

Reply:
You might want to use something like

```java
base
  .thenCompose(v -> role.getManager().setMentionable(true).submit())
  .thenCompose(v -> channel.sendMessage(message).submit())
  .thenCompose(v -> role.getManager().setMentionable(false).submit())
```

[`thenCompose` docs](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html#thenCompose-java.util.function.Function-)